EapWebInterface/src/main/java/de/hpi/unicorn/application/pages/input/generator/EventGenerator.java

```diff
@@ -121,278 +109,56 @@ public void generateEvents(int eventCount, int scaleFactor, EapEventType eventTy
+     * @param attributeInput contains the concerned event type attribute and its user input
      * @param eventValues the map containing already set values will be altered by reference
      */
-    private void setRandomValueFromRangeForAttribute(TypeTreeNode attribute, String proposedPossibleValues, Map<String, Serializable> eventValues) {
-        String possibleValues = proposedPossibleValues;
-
-        if (possibleValues == null || possibleValues.isEmpty()) {
-            possibleValues = defaultValues.get(attribute.getType());
-        }
-
-        switch (attribute.getType()) {
-            case STRING:
-                eventValues.put(attribute.getName(), getRandomStringFromInput(possibleValues));
-                break;
-            case INTEGER:
-                eventValues.put(attribute.getName(), getRandomIntFromInput(possibleValues));
-                break;
-            case FLOAT:
-                eventValues.put(attribute.getName(), getRandomFloatFromInput(possibleValues));
-                break;
-            case DATE:
-                eventValues.put(attribute.getName(), getRandomDateFromInput(possibleValues));
-                break;
-            default:
-                eventValues.put(attribute.getName(), "UNDEFINED");
-                break;
-        }
-    }
-
-    /**
-     * Returns the value of a "isInRange" function corresponding to the given type.
-     * This "class of functions" in general return a boolean indicating if the given input lies within a given range.
-     *
-     * @param input the input value to be tested to be in a range
-     * @param range the range the input value should be in
-     * @param type the type of the range (corresponds to the attribute type the value is conceived for)
-     * @return a boolean indicating if the input is within the range
-     */
-    private boolean isInRange(String input, String range, AttributeTypeEnum type) {
-        switch (type) {
-            case STRING:
-                return isInStringRange(input, range);
-            case INTEGER:
-                return isInIntegerRange(input, range);
-            case FLOAT:
-                return isInFloatRange(input, range);
-            case DATE:
-                return isInDateRange(input, range);
-            default:
-                return false;
-        }
-    }
-
-    /**
-     * Implements the "isInRange" function for a string range.
-     *
-     * @param input to be checked whether in range
-     * @param range to be searched in
-     * @return bool if string is in range
-     */
-    private boolean isInStringRange(String input, String range) {
-        String[] possibleValues = range.split(";");
-        for (String possibleValue : possibleValues) {
-            if (possibleValue.equals(input)) {
-                return true;
-            }
-        }
-        return false;
+    private void setRandomValueFromRangeForAttribute(AttributeInput attributeInput, Map<String, Serializable> eventValues) {
```

Comment:
yes, due to the AttributeInputs the name can be confusing.
I changed the signature with @20c3e5f.